src/libraries/System.Net.Mail/tests/Functional/MailAddressTest.cs

```diff
@@ -58,13 +60,21 @@ public static IEnumerable<object[]> GetInvalid_Address()
             yield return new object[] { "\"MrFo@Bar\"<>" };
             yield return new object[] { " " };
             yield return new object[] { "forbar" };
+            yield return new object[] { "" };
+            yield return new object[] { null };
         }
 
         [Theory]
         [MemberData(nameof(GetInvalid_Address))]
-        public void TestInvalidAddressInConstructo_Address(string invalidAddress)
+        public void TestInvalidAddressInConstructor_Address(string invalidAddress)
         {
-            Assert.Throws<FormatException>(() => new MailAddress(invalidAddress));
+            Action act = () => new MailAddress(invalidAddress);
+            if (invalidAddress is null)
+                Assert.Throws<ArgumentNullException>(act);
+            else if (invalidAddress == string.Empty)
+                Assert.Throws<ArgumentException>(act);
+            else
+                Assert.Throws<FormatException>(act);
```

Comment:
Just a personal taste, definitely won't hold the PR on this:
```suggestion
            Action act = () => new MailAddress(invalidAddress);
            if (invalidAddress is null)
            {
                Assert.Throws<ArgumentNullException>(act);
            }
            else if (invalidAddress == string.Empty)
            {
                Assert.Throws<ArgumentException>(act);
            }
            else
            {
                Assert.Throws<FormatException>(act);
            }
```